pkg/controller/seed-controller-manager/initial-application-installation-controller/controller.go

```diff
@@ -224,3 +240,12 @@ func (r *Reconciler) removeAnnotation(ctx context.Context, cluster *kubermaticv1
 	delete(cluster.Annotations, kubermaticv1.InitialApplicationInstallationsRequestAnnotation)
 	return r.Patch(ctx, cluster, ctrlruntimeclient.MergeFrom(oldCluster))
 }
+
+func getCiliumApplicationInstallation(userClusterClient ctrlruntimeclient.Client) (*appskubermaticv1.ApplicationInstallation, error) {
```

Comment:
I'd love for this to be more generic. I know that _at the moment_ it's only Cilium, but I could see us move more CNIs to the app infrastructure in the future.

I think this should be:

```suggestion
func getCNIApplicationInstallation(ctx context.Context, userClusterClient ctrlruntimeclient.Client, cniType kubermaticv1.CNIPluginType) (*appskubermaticv1.ApplicationInstallation, error) {
```

There could be a switch statement in here which only has a case for `kubermaticv1.CNIPluginTypeCilium` but could easily be extended later on.

Reply:
good point, let's do it beforehand